### PR TITLE
test: Cypress - Flaky Fix

### PR DESCRIPTION
**Pull request by Vijetha-Kaja**
## Description

**Fixed below flaky tests**
- Scrolling_Spec.ts
- API_Edit_spec.js
- MaintainContext&Focus_spec.js
 
## Type of change

- Flaky test fix

## How Has This Been Tested?
- Cypress test runs

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js [hunk 1]
```diff
@@ -1,60 +1,68 @@
...
-      _.agHelper.Sleep();
+      agHelper.Sleep();
       const { isPartialImport } = interception.response.body.data;
       if (isPartialImport) {
         // should reconnect modal
-        cy.get(reconnectDatasourceModal.SkipToAppBtn).click({
-          force: true,
+        cy.get("body").then(($ele) => {
+          if ($ele.find(reconnectDatasourceModal.SkipToAppBtn))
+            agHelper.GetNClick(reconnectDatasourceModal.SkipToAppBtn, 0, true);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js:25]: We need to handle that new modal click na now? or was it not needed? If not needed, where is that modal closed please?

File: app/client/cypress/support/ApiCommands.js [hunk 2]
```diff
@@ -111,9 +111,19 @@ Cypress.Commands.add(
...
+    agHelper.Sleep();
+
+    cy.get("body").then(($ele) => {
+      if ($ele.find(ApiEditor.apiResponseObject).length <= 0) {
+        agHelper
+          .GetElement(ApiEditor.apiDebuggerLink)
+          .last()
+          .contains(apiName)
+          .click();
+      }
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/ApiCommands.js:123]: What this does please?

File: app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js (hunk 1 above)
Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js:25]: oh yes! sorry forgot about that!

File: app/client/cypress/support/ApiCommands.js (hunk 2 above)
Comment (Vijetha-Kaja) [reply app/client/cypress/support/ApiCommands.js:123]: this api response is not expanding sometimes in logs , so to expand that

Comment (Aishwarya-U-R) [reply app/client/cypress/support/ApiCommands.js:123]: Ah ok!

File: app/client/cypress/limited-tests.txt [hunk 3]
```diff
@@ -2,4 +2,7 @@
 #cypress/e2e/**/**/*
 
 # To run only limited tests - give the spec names in below format:
-cypress/e2e/Regression/ClientSide/Widgets/Dropdown/Dropdown_onOptionChange_spec.js
+#cypress/e2e/Regression/ClientSide/Widgets/Dropdown/Dropdown_onOptionChange_spec.js
```
Comment (Aishwarya-U-R) [inline app/client/cypress/limited-tests.txt:5]: Please revert this file changes. Can merge this PR.

**Review by Aishwarya-U-R — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js [hunk 4]
```diff
@@ -1,60 +1,70 @@
 import reconnectDatasourceModal from "../../../../locators/ReconnectLocators";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/IDE/MaintainContext&Focus_spec.js:1]: @Vijetha-Kaja - Please also fix formatting in this file.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**